synfig-studio/src/gui/canvasview.cpp

```diff
@@ -577,7 +577,7 @@ CanvasView::CanvasView(etl::loose_handle<Instance> instance,etl::handle<CanvasIn
 
 	//info("Canvasview: Before big chunk of allocation and tabling stuff");
 	//create all allocated stuff for this canvas
-	Gtk::Alignment *widget_space = Gtk::manage(new Gtk::Alignment());
+	Gtk::Fixed *widget_space = Gtk::manage(new Gtk::Fixed);
```

Comment:
If a widget is really necessary, what about just a plain Widget?

Reply:
We have `set_margin_start()` and `set_margin_end()` to achieve the same behavior. So instead of placing that space widget, find out where it is placed and determine if our next widget is proceeding or succeeding from space widget, then use the appropriate margin property.

Reply:
I used Gtk::Fixed because I was thinking that it is the purpose of Fixed is to showcase that something related to arrangement is going on in the code which contains this class! If its not good, then I will change that to whatever needed ! No issues :)

Reply:
`Fixed` is a container that places children in fixed coordinates (they don't change if you expand/shrink the container dimensions).

Reply:
@rodolforg I am unable to find where canvaspreview is ? I want to try gtkmm container widget classes on it ! and it can't be done without trial and error (atleast for me 🙂). It would be great if you guys can tell me that how can I find some piece of code in synfig studio. As it will also help in the later stages as well !

Thanks and sorry for inconvenience. I will now keep on replacing alignment.h wherever there is no chance of conflicts for the time being.

Reply:
@DhairyaBahl `widget_space` is added here
https://github.com/synfig/synfig/blob/9b6c3a2a62796647f0937c6ad8354f1411c8776c/synfig-studio/src/gui/canvasview.cpp#L594-L596
You can notice it's trying to add 4 spaces between the canvas workspace and top/right toolbars. Delete the `widget_space` along with the `Gtk::Alignment` code and replace them with `widget_right_bar->set_margin_start(4)` (Check if this is really needed) and `widget_top_bar->set_margin_bottom(4)`. The ruler changes a bit if you try to use `widget_work_area->set_margin_top(4)` so I used top bar widget there.

You can use pen and paper to visually draw the layout determining how widgets are packed and finally decide which margin property to use `start/end/top/bottom`. Hope you got enough idea.

Reply:
I pushed my latest changes after making the required changes ! Thanks for helping with the grid ! after drawing it in my notebook, I was able to see a clear picture of the grid. 

I also wanted to ask how can I find particular code in the synfig-studio after building so that I can make changes and build code again and again and experiment with the code as much as possible. Kindly tell me how to do so. Sometimes I am able to find a part of code in synfig-studio and sometimes I fail to do so. Kindly tell me the best approach to do so.

Thanks and Regards :)

Reply:
> I also wanted to ask how can I find particular code in the synfig-studio after building so that I can make changes and build code again and again and experiment with the code as much as possible.

That depends on what you are talking about ;)
I listed some info about synfig code folder structure here:
https://forums.synfig.org/t/help-required-in-understanding-the-synfig-studio-code/10505/4

You can use `git grep` for searching some words inside project source code.

But if you are 'attacking' a specific file for now, I can't understand your question.

Reply:
Okkayy ! I got your point ! thanks for help !  Nope I meant the same but I was unable to frame the question lol ! my bad ! anyways thanks again